### PR TITLE
refactor zip viewer to class vue and typescript

### PR DESCRIPTION
**Pull request by HNOONa-0**
# Description

Fixes: #7692

# Checklist:

- [x] The code follows the [coding guidelines](https://github.com/omegaup/omegaup/wiki/Coding-guidelines) of omegaUp.
- [x] The tests were executed and all of them passed.
- [ ] If you are creating a feature, the new tests were added.

### CODE REVIEW

File: frontend/www/js/omegaup/graderv2/ZipViewerComponent.vue [hunk 1]
```diff
@@ -0,0 +1,76 @@
+<template>
```
Comment (HNOONa-0) [inline frontend/www/js/omegaup/graderv2/ZipViewerComponent.vue:1]: https://github.com/omegaup/omegaup/blob/2bc420a29849fb5c354534ecf29d6335eb00a07a/frontend/www/js/omegaup/grader/ZipViewerComponent.vue#L1-L77

File: frontend/www/js/omegaup/graderv2/ZipViewerComponent.vue [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  <div
+    class="root d-flex flex-row h-100 bg-dark text-white"
```
Comment (pabo99) [inline frontend/www/js/omegaup/graderv2/ZipViewerComponent.vue:3]: nit:
```suggestion
    class="root d-flex flex-row h-100"
```

File: frontend/www/js/omegaup/graderv2/ZipViewerComponent.vue [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    }"
+  >
+    <div class="filenames">
+      <div class="list-group">
+        <button
+          v-if="!zip"
+          class="list-group-item list-group-item-action disabled"
+          type="button"
+        >
+          <em>Empty</em>
```
Comment (pabo99) [inline frontend/www/js/omegaup/graderv2/ZipViewerComponent.vue:16]: Could you add this string to the translation files?

File: frontend/www/js/omegaup/graderv2/ZipViewerComponent.vue [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+  select(item: JSZipObject): void {
+    this.active = item.name;
+    item
+      .async('string')
+      .then((value: string) => {
+        this.contents = value;
+      })
+      .catch(Util.asyncError);
+  }
```
Comment (pabo99) [inline frontend/www/js/omegaup/graderv2/ZipViewerComponent.vue:57]: Will it be possible move this content to the `.ts` file and here emit an acttion?

Comment (HNOONa-0) [reply frontend/www/js/omegaup/graderv2/ZipViewerComponent.vue:59]: Sure! After typescript store is fully merged we can move logic over.

Comment (pabo99) [reply frontend/www/js/omegaup/graderv2/ZipViewerComponent.vue]: Okay.

For now, we can approve this PR

**Review by pabo99 — APPROVED** [review]
LGTM

**Review by heduenas — APPROVED** [review]

**Pull request merged**